src/object/cli_obj.c

```diff
@@ -1487,6 +1535,7 @@ dc_obj_update(tse_task_t *task)
 		shard_arg = tse_task_buf_embedded(shard_task,
 						  sizeof(*shard_arg));
 		shard_arg->epoch		= epoch;
+		daos_generate_dti(&shard_arg->dti);
```

Comment:
just a question that for client-side replica mode (now it is still supported if set DAOS_IO_SRV_DISPATCH as zero) still need to generate the dti and other related dtx handling? (seem no? then in client side replica mode maybe can simplify some code, but can be done later.)

Reply:
Good question! I will update the patch to make the client to generate empty "dti", then the server side logic can detect that properly.